File: modules/yandex-map/android/src/main/java/expo/modules/yandexmap/modules/PolygonModule.kt
package expo.modules.yandexmap.modules

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import expo.modules.yandexmap.model.Coordinate
import expo.modules.yandexmap.view.PolygonView

class PolygonModule : Module() {
    override fun definition() = ModuleDefinition {
        Name("Polygon")

        View(PolygonView::class) {
            Prop("points") { view: PolygonView, points: List<Coordinate> ->
                view.setPointsValue(points)
            }

            Prop("innerPoints") { view: PolygonView, points: List<Coordinate> ->
                view.setInnerPointsValue(points)
            }

            Prop("strokeColor") { view: PolygonView, color: String ->
                view.setStrokeColorValue(color)
            }

            Prop("fillColor") { view: PolygonView, color: String ->
                view.setFillColorValue(color)
            }

            Prop("strokeWidth") { view: PolygonView, width: Float ->
                view.setStrokeWidthValue(width)
            }
        }
    }
}

File: modules/yandex-map/ios/PolygonModule.swift
import ExpoModulesCore

public class PolygonModule: Module {
    public func definition() -> ModuleDefinition {
        Name("Polygon")
        
        View(PolygonView.self) {
            Prop("points") { (view: PolygonView, points: Array<Coordinate>) in
                view.setPointsValue(points)
            }
            
            Prop("innerPoints") { (view: PolygonView, points: Array<Coordinate>) in
                view.setInnerPointsValue(points)
            }
            
            Prop("strokeColor") { (view: PolygonView, color: String) in
                view.setStrokeColorValue(color)
            }
            
            Prop("strokeWidth") { (view: PolygonView, width: Float) in
                view.setStrokeWidthValue(width)
            }
            
            Prop("fillColor") { (view: PolygonView, color: String) in
                view.setFillColorValue(color)
            }
        }
    }
}

